pkgs/development/compilers/dart/package-source-builders/file_picker/default.nix

```
@@ -0,0 +1,25 @@
+{
+  stdenv,
+  zenity,
+}:
+
+{ version, src, ... }:
+
+stdenv.mkDerivation rec {
+  pname = "file_picker";
+  inherit version src;
+  inherit (src) passthru;
+
+  postPatch = ''
+    substituteInPlace lib/src/linux/file_picker_linux.dart \
+        --replace-fail "isExecutableOnPath('zenity')" "isExecutableOnPath('${zenity}/bin/zenity')"
```

Side note: this hack works because isExecutableOnPath uses `which`, and `which` returns the path itself if it's given a full path.
https://github.com/miguelpruivo/flutter_file_picker/blob/4dc10ccf3d703b92d62f0cc812e71989fd25d58e/lib/src/utils.dart#L57

---

Wait, does
```suggestion
        --replace-fail "isExecutableOnPath('zenity')" "'${zenity}/bin/zenity'"
```

work?